types/Circus.js

```diff
@@ -9,7 +9,7 @@
 
 export type DoneFn = (reason?: string | Error) => void;
 export type BlockFn = () => void;
-export type BlockName = string | Function;
+export type BlockName = string;
```

Comment:
Loos like we can simplify this type now? https://github.com/facebook/jest/blob/c4fba2d9efe04aed7114b27feacd3453ff9fd4f7/types/Circus.js#L158
And maybe change the misleading `testPath` to `testTitle`

Reply:
`testPath` is an artefact from `jasmine`, not sure we can change it?

I like the expressiveness of `Array<TestName | BlockName>`, even though `Array<string>` is also correct.